cupy/random/_generator_api.pyx

```diff
@@ -219,6 +222,49 @@ class Generator:
         # omitted args.
         return (<object>y).astype(dtype, copy=False)
 
+    def gamma(self, shape, scale=1.0, size=None):
+        """Returns an array of samples drawn from a gamma distribution.
+
+        .. seealso::
+            - :func:`cupy.random.gamma` for full documentation
+            - :meth:`numpy.random.RandomState.gamma`
+        """
+        return self.standard_gamma(shape, size) * scale
+
+    def standard_gamma(self, shape, size=None, dtype=numpy.float64, out=None):
+        """Returns an array of samples drawn from a standard gamma distribution.
+
+        .. seealso::
+            - :func:`cupy.random.standard_gamma` for full documentation
+            - :meth:`numpy.random.RandomState.standard_gamma`
+        """
+        cdef ndarray y
+
+        if shape < 0.0:
+            raise ValueError('shape < 0')
+
+        if out is None or out.dtype.char != 'd':
+            y = ndarray(size if size is not None else (), numpy.float64)
+
+        if out is not None:
+            self._check_output_array(dtype, size, out)
+            if out.dtype.char == 'd':
+                y = out
```

Comment:
How about this?
```suggestion
        y = None
        
        if out is not None:
            self._check_output_array(dtype, size, out)
            if out.dtype.char == 'd':
                y = out
        if y is None:
             y = ndarray(size if size is not None else (), numpy.float64)
```
Then you can remove line 258, which mistakenly allocates a new `y` regardless of `out`.

Reply:
Ah, good catch! will update it tomorrow :)